src/utils/searchHelper.ts

```diff
@@ -43,5 +47,103 @@ export function getSearchWith(
     });
 
   // we return a string to use it inside links
+
   return newParams.toString();
 }
+
+export const getFilteredPeople = (
+  searchParams: URLSearchParams,
+  peopleFS: Person [] | null,
+) => {
+  if (!peopleFS) {
+    return null;
+  }
+
+  const currentSex = searchParams.get('sex');
+  const currentSearch = searchParams.get('search');
+  const currentCenturies = searchParams.getAll('century');
+  const currentSort = searchParams.get('sort');
+  const currentOrder = searchParams.get('order');
```

Comment:
The helper function shouldn't know anything about search params, just accepts filter/sorting criteria and returns a new array in accordance.